pyccel/ast/numpy_wrapper.py

```diff
@@ -0,0 +1,289 @@
+#------------------------------------------------------------------------------------------#
+# This file is part of Pyccel which is released under MIT License. See the LICENSE file or #
+# go to https://github.com/pyccel/pyccel/blob/master/LICENSE for full license details.     #
+#------------------------------------------------------------------------------------------#
+# pylint: disable=missing-function-docstring
+
+"""
+Handling the transitions between python code and C code.
+"""
+
+import numpy as np
+
+from .datatypes         import NativeInteger, NativeReal, NativeComplex
+from .datatypes         import NativeBool, NativeGeneric, NativeVoid
+
+from .cwrapper          import PyccelPyObject, PyccelPyArrayObject
+
+from .core              import FunctionDef, FunctionCall
+
+from .literals          import LiteralInteger
+
+from .variable          import Variable
+
+from ..errors.errors   import Errors
+
+errors = Errors()
+
+__all__ = (
+    'numpy_get_ndims',
+    'numpy_get_data',
+    'numpy_get_dim',
+    'numpy_get_stride',
+    'numpy_check_flag',
+    'numpy_get_base',
+    'numpy_itemsize',
+    'numpy_flag_own_data',
+    'numpy_flag_c_contig',
+    'numpy_flag_f_contig',
+    'numpy_dtype_registry',
+    'PyArray_CheckScalar',
+    'PyArray_ScalarAsCtype',
+)
+
+#-------------------------------------------------------------------
+#                      Numpy functions
+#-------------------------------------------------------------------
+# All the numpy function list are part of  numpy/c api
+# https://numpy.org/doc/stable/reference/c-api/array.html
```

Comment:
Are these functions still used? Or is everything handled in cwrapper.h/cwrapper.c?

Reply:
not all of them
```
numpy_get_type
PyArray_ScalarAsCtype
```
are still used